### PR TITLE
Add dependabot to update GH Actions and Python dependencies

### PR DESCRIPTION
**Pull request by pnacht**
Hey, it's Pedro (see https://github.com/keras-team/keras-nlp/pull/1305 and https://github.com/keras-team/keras-nlp/pull/1319) and I'm back with another security suggestion.

This PR is identical to the ones I sent to [Keras](https://github.com/keras-team/keras/pull/18834) and [KerasCV](https://github.com/keras-team/keras-cv/pull/2259). It configures Dependabot to monitor the GitHub Actions used in KerasNLP's workflows, as well as its Python dependencies.

I've configured Dependabot to send a single monthly PR (every 1st of the month) updating all dependencies in each ecosystem (see the PRs in my fork: https://github.com/pnacht/keras-nlp/pull/1 and https://github.com/pnacht/keras-nlp/pull/2).

I have taken the liberty of merging those dependabot PRs into this one so you don't receive such PRs right after merging this one.

Note that Dependabot will also update the `tf-nightly`, `tf-nightly-cpu` and `tensorflow-text-nightly` Python dependencies to the latest nightly snapshot. This will ensure you're testing on more recent versions of these unreleased dependencies. However, if you're concerned about updating to a broken nightly (which would likely be detected by failing tests on the Dependabot PR), I can configure Dependabot to ignore those dependencies so you can update them manually if you prefer.

(Following https://github.com/keras-team/keras/issues/18833#issuecomment-1828743533, I haven't sent an issue for this. Let me know if KerasNLP prefers always having an accompanying issue to discuss my contributions).

### CODE REVIEW

**Review by mattdangerw — APPROVED** [review]
LGTM! @sampathweb do you know why GPU tests are not running? Would be good to check these as we are updating our requirements files.

**Comment by sampathweb** [comment]
> LGTM! @sampathweb do you know why GPU tests are not running? Would be good to check these as we are updating our requirements files.

Right now, GPU tests run when we - 

* Add "Kokoro-force-run" label manually.  I just did for this PR so it should run.
* When we merge the PR to master (Runs on mater branch automatically on the merged commit).
I have a pending activity to add this tag (via a GH Action) automatically on PR approval.  But have not done it so far.  I will look into it.

File: requirements-jax-cuda.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Tensorflow cpu-only version.
-tf-nightly-cpu==2.16.0.dev20231109  # Pin a working nightly until rc0.
-tensorflow-text-nightly==2.16.0.dev20231109  # Pin a working nightly until rc0.
+tf-nightly-cpu==2.16.0.dev20231227  # Pin a working nightly until rc0.
```
Comment (sampathweb) [inline requirements-jax-cuda.txt:2]: Is it possible to sync the nightly version across all Keras eco system - Keras, NLP and CV.  Currently `keras` is pinned to 11/30 and its on different schedule from dependabot. Can we sync the dependabot to update all three packages on same date? - 
https://github.com/keras-team/keras/blob/master/requirements-tensorflow-cuda.txt

Comment (pnacht) [reply requirements-jax-cuda.txt:2]: So, this will actually help keep these in sync!

When Dependabot is configured to run monthly (as I've done for all these projects), it always runs on the 1st of each month. So all projects will receive Dependabot PRs updating these nightlies to the last day of the previous month. Assuming all projects merge those PRs, they'll always run on the same recent versions.

---

Note that if this PR is merged before the end of the year, you'll immediately receive a Dependabot PR updating the nightlies to the latest version, and then receive another PR soon after, on January 1st. This actually happened in /keras and momentarily confused the maintainers, since they:

- merged my PR setting up Dependabot (https://github.com/keras-team/keras/pull/18834) on November 30th
- immediately got a Dependabot PR (https://github.com/keras-team/keras/pull/18856) updating the nightlies
- and then got another Dependabot PR (https://github.com/keras-team/keras/pull/18865) updating the nightlies the very next day (December 1st).

Comment (sampathweb) [reply requirements-jax-cuda.txt:2]: LGTM.

**Review by sampathweb — APPROVED** [review]
Please rebase with the master since the requirements files hv been updated.

**Comment by pnacht** [comment]
Done.

**Comment by sampathweb** [comment]
@pnacht - Looks like these commits are very old.  Can you rebase once more?

**Comment by pnacht** [comment]
@sampathweb done.
